runatlantis.io/blog/4-reasons-to-try-hashicorps-new-free-terraform-remote-state-storage.md

```diff
@@ -0,0 +1,157 @@
+    ...
+    "resources": {
+      "aws_instance.web": {
+        "type": "aws_instance",
+        "primary": {
+          "id": "i-0ad17607e5ee026d0",
+     ...
+}
+```
+
+Here you can see that the resource `aws_instance.web` from our Terraform code is mapped to the instance ID `i-0ad17607e5ee026d0`.
+
+So if Terraform state is just a file, then what is remote state?
+
+## Remote State
+
+By default, Terraform writes its state file to your local filesystem. This is okay for personal projects, but once you start working with a team, things get messy. In a team, you need to make sure everyone has an up to date version of the state file **and** ensure that two people aren't making concurrent changes.
+
+Enter remote state! Remote state is just storing the state file remotely, rather than on your filesystem. With remote state, there's only one copy so Terraform can ensure you're always up to date. To prevent team members from modifying state at the same time, Terraform can lock the remote state.
+
+> Remote state is just storing the state file remotely, rather than on your filesystem.
+
+Alright, so remote state is great, but unfortunately setting it up can be a bit tricky. In AWS, you can store it in an S3 bucket, but you need to create the bucket, configure it properly, set up its permissions properly, create a DynamoDB table for locking and then ensure everyone has proper credentials to write to it. It's much the same story in the other clouds.
+
+As a result, setting up remote state can be an annoying stumbling block as teams adopt Terraform.
+
+This brings us to the first reason to try HashiCorp's Free Remote State Storage...
+
+## Reason #1 — Easy To Set Up
+
+Unlike other remote state solutions that require complicated setup to get right, setting up free remote state storage is easy.
+
+> Setting up HashiCorp's free remote state storage is easy
+
+Step 1 — Sign up for your [free Terraform Cloud](https://app.terraform.io/signup) account
+
+Step 2 — When you log in, you'll land on this page where you'll create your organization:
+
+![](/blog/4-reasons-to-try-hashicorps-new-free-terraform-remote-state-storage/pic1.webp)
+
+Step 3 — Next, go into User Settings and generate a token:
+
+![](/blog/4-reasons-to-try-hashicorps-new-free-terraform-remote-state-storage/pic2.webp)
+
+Step 4 — Take this token and create a local ~/.terraformrc file:
+
+```tf
+credentials "app.terraform.io" {
+  token = "mhVn15hHLylFvQ.atlasv1.jAH..."
+}
+```
+
+Step 5 — That's it! Now you're ready to store your state.
+
+In your Terraform project, add a `terraform` block:
+
+```tf{3,5}
+terraform {
+  backend "remote" {
+    organization = "my-org" # org name from step 2.
+    workspaces {
+      name = "my-app" # name for your app's state.
+    }
+  }
+}
+```
+
+Run `terraform init` and tada! Your state is now being stored in Terraform Enterprise. You can see the state in the UI:
+
+![](/blog/4-reasons-to-try-hashicorps-new-free-terraform-remote-state-storage/pic3.webp)
```

Comment:
I don't know if VitePress supports it, but I _really_ love the pattern of having images side by side with the markdown in a folder, and then copy it to the public directory at build time; prevents them from being spread across a bunch of different directories as the blog grows

Reply:
yeah, currently it sits in the public folder, what do you think of it?

Reply:
I don't think its blocking for merging this; but something we could look into improving down the line; it should affect public URLs if we change it later

Reply:
yeah, feel free to raise an issue or chat over slack.